pkg/drivers/vmnet/vmnet.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	var info interfaceInfo
+	if err := json.NewDecoder(stdout).Decode(&info); err != nil {
+		return fmt.Errorf("failed to decode vmnet interface info: %w", err)
+	}
+
+	log.Infof("Got mac address %q", info.MACAddress)
+	h.macAddress = info.MACAddress
+
+	return nil
+}
+
+// GetMACAddress reutuns the mac address assigned by vment framework.
+func (h *Helper) GetMACAddress() string {
+	return h.macAddress
+}
+
+// Stop the vmnet-helper child process.
+func (h *Helper) Stop() error {
+	path := h.machinePath(pidfileName)
+	pid, err := readPidfile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+	log.Infof("Terminate vmnet-helper (pid=%v)", pid)
+	// Terminate sudo, which will terminate vmnet-helper.
+	if err := process.Signal(syscall.SIGTERM); err != nil {
+		if err != os.ErrProcessDone {
+			return err
+		}
+	}
+	os.Remove(path)
+	return nil
+}
+
+// Kill the vmnet-helper child process.
+func (h *Helper) Kill() error {
+	path := h.machinePath(pidfileName)
+	pid, err := readPidfile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+	log.Infof("Kill vmnet-helper process group (pgid=%v)", pid)
+	if err := syscall.Kill(-pid, syscall.SIGKILL); err != nil {
+		if err != syscall.ESRCH {
+			return err
+		}
+	}
+	os.Remove(path)
+	return nil
+}
+
+// GetState returns the vment-helper child process state.
+func (h *Helper) GetState() (state.State, error) {
+	path := h.machinePath(pidfileName)
+	pid, err := readPidfile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return state.Stopped, nil
+		}
+		return state.Error, err
+	}
+	if err := checkPid(pid); err != nil {
+		// No pid, remove pidfile
+		os.Remove(path)
+		return state.Stopped, nil
+	}
+	return state.Running, nil
+}
+
+func (h *Helper) openLogfile() (*os.File, error) {
+	path := h.machinePath(logfileName)
+	return os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o600)
+}
+
+func (h *Helper) machinePath(fileName string) string {
+	return filepath.Join(h.StorePath, "machines", h.MachineName, fileName)
+}
+
+func writePidfile(path string, pid int) error {
+	data := fmt.Sprintf("%v", pid)
+	if err := os.WriteFile(path, []byte(data), 0600); err != nil {
+		return err
+	}
+	return nil
+}
+
+func readPidfile(path string) (int, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return -1, err
+	}
+	pid, err := strconv.Atoi(strings.TrimSpace(string(data)))
+	if err != nil {
+		return -1, err
+	}
+	return pid, nil
+}
+
+func checkPid(pid int) error {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+	return process.Signal(syscall.Signal(0))
+}
+
+// uuidFromName generated a random UUID (type 4) from string name. This is
+// useful for creating interface ID from virtual machine name to ensure the same
+// MAC address in all runs.
+func uuidFromName(name string) string {
```

Comment:
I'll remove this for now since other drivers use new mac address for new machine, regardless of the machine name.